CMakeLists.txt

```diff
@@ -97,28 +97,10 @@ include(${PROJECT_SOURCE_DIR}/cmake/dependencies.cmake)
 add_library(umd_common_directories INTERFACE)
 target_include_directories(umd_common_directories INTERFACE ${PROJECT_SOURCE_DIR} ${PROJECT_SOURCE_DIR}/device ${PROJECT_SOURCE_DIR}/third_party/fmt/include)
 
-if(NOT DEFINED ENV{ARCH_NAME})
-    message(FATAL_ERROR "Please set ARCH_NAME to grayskull, wormhole_b0, or blackhole")
-elseif($ENV{ARCH_NAME} STREQUAL "grayskull")
-    message(STATUS "UMD: Building for Grayskull")
-    target_include_directories(umd_common_directories INTERFACE 
-        ${PROJECT_SOURCE_DIR}/device/grayskull 
-        ${PROJECT_SOURCE_DIR}/src/firmware/riscv/grayskull
-    )
-elseif($ENV{ARCH_NAME} STREQUAL "wormhole_b0")
-    message(STATUS "UMD: Building for Wormhole")
-    target_include_directories(umd_common_directories INTERFACE 
-        ${PROJECT_SOURCE_DIR}/device/wormhole 
-        ${PROJECT_SOURCE_DIR}/src/firmware/riscv/wormhole
-    )
-elseif($ENV{ARCH_NAME} STREQUAL "blackhole")
-    message(STATUS "UMD: Building for Blackhole")
-    target_include_directories(umd_common_directories INTERFACE 
-        ${PROJECT_SOURCE_DIR}/device/blackhole 
-        ${PROJECT_SOURCE_DIR}/src/firmware/riscv/blackhole
-    )
-endif()
-
 add_subdirectory(${PROJECT_SOURCE_DIR}/device)
-add_subdirectory(${PROJECT_SOURCE_DIR}/tests EXCLUDE_FROM_ALL)
+
+option(TT_UMD_BUILD_TESTS "Enables build of tt_umd tests" OFF)
+if(TT_UMD_BUILD_TESTS)
+    add_subdirectory(tests)
```

Comment:
nit: ${PROJECT_SOURCE_DIR}/tests

Reply:
I was of the reverse inclination, because with CMake ‘add_subdirectory’ it is implied that tests is a path/subdirectory relative to the current CMakeLists.txt

This method is more flexible if the folder structure changes.

but it’s not a big deal.

Let me know if you insist and I’ll change it

Reply:
I just complied to avoid the extra back and forth.